frontend/src/layout/navigation-3000/components/SidebarList.tsx

```diff
@@ -537,23 +614,81 @@ function SidebarListItemSkeleton({ style }: { style: React.CSSProperties }): JSX
     return (
         <li
             className="SidebarListItem SidebarListItem__link"
-            style={style} // eslint-disable-line react/forbid-dom-props
+            // eslint-disable-next-line react/forbid-dom-props
+            style={style}
         >
             <LemonSkeleton />
         </li>
     )
 }
 
-function SidebarListItemAccordion({ category }: { category: ListItemAccordion }): JSX.Element {
+function SidebarListItemAccordion({
+    category,
+    onEditing,
+}: {
+    category: ListItemAccordion
+    onEditing: (isEditing: boolean) => void
+}): JSX.Element {
     const { listItemAccordionCollapseMapping } = useValues(navigation3000Logic)
     const { toggleListItemAccordion } = useActions(navigation3000Logic)
+    const [isRenaming, setIsRenaming] = useState(false)
+    const [newName, setNewName] = useState(category.name || capitalizeFirstLetter(pluralizeCategory(category.noun)))
+    const [isSaving, setIsSaving] = useState(false)
+
+    const ref = useRef<HTMLDivElement>(null)
 
-    const { key, items } = category
+    const { key } = category
 
-    const isEmpty = items.length === 0
     const keyString = Array.isArray(key) ? key.join(ITEM_KEY_PART_SEPARATOR) : key.toString()
     const isExpanded = !(keyString in listItemAccordionCollapseMapping) || !listItemAccordionCollapseMapping[keyString]
 
+    const startRenaming = useCallback((): void => {
+        if (category.onRename) {
+            setIsRenaming(true)
+        }
+    }, [category.onRename])
+
+    const cancelRenaming = (): void => {
+        setIsRenaming(false)
+        setNewName(category.name || capitalizeFirstLetter(pluralizeCategory(category.noun)))
+    }
+
+    const saveNewName = async (): Promise<void> => {
+        if (category.onRename && newName !== category.name) {
+            setIsSaving(true)
+            try {
+                await category.onRename(newName)
+            } catch (error) {
+                captureException(error)
+                lemonToast.error('Could not rename item')
+            } finally {
+                setIsSaving(false)
+                setIsRenaming(false)
+            }
+        } else {
+            cancelRenaming()
+        }
+    }
+
+    useEffect(() => {
+        if (category.onRename && !isRenaming) {
+            const onDoubleClick = (): void => {
+                startRenaming()
+            }
+            const element = ref.current
+            if (element) {
+                element.addEventListener('dblclick', onDoubleClick)
+                return () => {
+                    element.removeEventListener('dblclick', onDoubleClick)
+                }
+            }
+        }
+    }, [isRenaming, category.onRename, startRenaming])
```

Comment:
There is some propagation behavior on this where clicking once toggles the openness of the folder and the double click is what is needed for editing. I ran into some of this with the tab renaming, maybe it would best to discuss a system to handle this kind of thing generally, with a consistent edit state that isn't too disruptive.

Reply:
yeah agreed. will roll with this for now but will need to revisit the interaction behaviors